src/CommonProviderImplementation/Helpers.fs

```diff
@@ -74,6 +74,7 @@ type DisposableTypeProviderForNamespaces() as x =
     inherit TypeProviderForNamespaces()
   
     let disposeActions = ResizeArray()
+    let dictLock = obj()
```

Comment:
The issue with locking directly an object, is that if that object is accessible outside some external code may lock it as well, causing deadlocks. That's not the case here, as disposeActions is a local value binding, so please revert